src/Compilers/Core/Portable/CommandLine/AnalyzerConfigOptionKeys.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Holds keys for all options in an <see cref="AnalyzerConfigSet"/>.
+    /// </summary>
+    public readonly struct AnalyzerConfigOptionKeys
```

Comment:
going to bring to API review i presume?

Reply:
Yes, this is just for experimental purpose. I want to first verify all tests pass (hence the draft PR), then measure perf and then bring it for API review. Finally, mark the PR ready for review. Feel free to review the changes now if you'd like though.